### PR TITLE
docs: Update proposal 2 with design changes

### LINKED ISSUE #117: [TRACKING] Implement Proposal 2: Run
Issue to track the implementation of Proposal 2: Run - Call and define benchmark tests as GitHub Action jobs & workflows

Issue tracking template copied from https://github.com/cncf-tags/green-reviews-tooling/issues/101 by @rossf7 :)

# Overview

Implement Run step of the Green Review pipeline.

- Parent Issue https://github.com/cncf-tags/green-reviews-tooling/issues/83
- Proposal 2 contains the design https://github.com/cncf-tags/green-reviews-tooling/blob/main/docs/proposals/proposal-002-run.md

## Implementation Tasks

1 and 2 can be worked in parallel. 3 is blocked by 1 & 2.

### 1. Add benchmark components to the pipeline
- [x] #118

### 2. Refactor Falco tests into a benchmark workflow & jobs
- [ ] #121 

### 3. Once everything is running, ensure reproducibility
- [ ] Delete benchmark job artefacts after each run
- [ ] Add a template for how to add a benchmark workflow + job

### PR DESCRIPTION
**Pull request by rossf7**
#### What type of PR is this?

kind/documentation

#### What this PR does / why we need it:

Updates proposal 2 with design changes implemented in #127 

#### Which issue(s) this PR fixes:

Fixes #117

#### Special notes for your reviewer (optional):

Changes are quite big so sorry the diff is hard to follow. 
I also changed the diagram to mermaid based on the diagram @dipankardas011 added to proposal 1.

### CODE REVIEW

File: docs/proposals/proposal-002-run.md [hunk 1]
```diff
@@ -164,108 +165,134 @@ change are understandable. This may include manifests or workflow examples
...
 
-The benchmark job has some test instructions/steps. In this case, it applies an upstream Kubernetes manifest. This manifest contains a `while` loop that runs `stress-ng`. The manifest already defines where the test should run in the cluster i.e. in which namespace. The functional unit test is time-bound in this case and scoped to 15 minutes. Therefore, we deploy this test, wait for 15 minutes, then delete the manifest to end the loop. The test steps depend on the functional unit of each CNCF project.
+### Versioning / Security
 
-In the example above, the Kubernetes manifest that is applied to the cluster is located in a different repository: this is the case of an externally defined benchmark
-
-Each workflow should ensure that any artefacts that were deployed as part of the benchmark job should be deleted at the end of the test run.
+Manifests in `project.json` are pinned to a Git commit SHA rather than a branch such as `main`. This mitigates the risk that a malicious workload could be included in the benchmark manifest and ensures that any changes to the manifests are reviewed by one of the Green Reviews maintainers.
```
Comment (rossf7) [inline docs/proposals/proposal-002-run.md:290]: @leonardpahlke @nikimanoledaki @dipankardas011 FYI I added this note on pinning manifests to SHAs following our discussion in last weeks meeting.

LMK if more detail is needed

Comment (dipankardas011) [reply docs/proposals/proposal-002-run.md:290]: I think we should add some seccomp profiles, quota limits on the namespace we are dealing with so that we don't have impact on other monitoring workloads, also we should perform some sort of image scaning before begining to start benchmarking to safeguard from any potential problems

anyways let me know your thoughts :)
I was just checking out CIS benchmark for any k8s cluster

**Review by dipankardas011 — APPROVED** [review]
apart from the above comment LGTM

File: docs/proposals/proposal-002-run.md (hunk 1 above)
Comment (rossf7) [reply docs/proposals/proposal-002-run.md:290]: @dipankardas011 Those are good recommendations for securing the cluster but I think we should keep this focused on the pipeline.

The future of the cluster is uncertain. If we only provision on-demand the design is likely to be different to a permanent cluster even a single node one. So I would not invest time until the design is clear.

However your approach of being guided by the CIS benchmark is sound. It's also a good point that by running both the project components and the benchmarks in the `benchmark` namespace we can restrict its access.

**Comment by rossf7** [comment]
Sorry I closed the wrong PR 🤦‍♂️

**Review by nikimanoledaki — APPROVED** [review]
LGTM, thank you @rossf7 !!
